File: Artist/app/src/main/java/com/example/artist/ui/topartistslisting/di/TopArtistsComponent.java
package com.example.artist.ui.topartistslisting.di;


import com.example.artist.ui.topartistslisting.TopArtistsFragment;

import javax.inject.Singleton;

import dagger.Component;


@Singleton
@Component(modules = {TopArtistsModule.class})
public interface TopArtistsComponent {
    void inject(TopArtistsFragment target);
}

File: Artist/app/src/main/java/com/example/artist/ui/topartistslisting/di/TopArtistsComponent.kt
package com.example.artist.ui.topartistslisting.di

import com.example.artist.ui.topartistslisting.TopArtistsFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [TopArtistsModule::class])
interface TopArtistsComponent {
    fun inject(target: TopArtistsFragment)
}